pennylane_cirq/cirq_device.py

```diff
@@ -167,58 +159,56 @@ def operations(self):
+    def apply(self, operations, rotations=None, **kwargs):
+        rotations = rotations or []
 
-        # This code is adapted from the pennylane-qiskit plugin
-        for e in self.obs_queue:
-            # Identity and PauliZ need no changes
-            if e.name == "PauliX":
-                # X = H.Z.H
-                self.apply("Hadamard", wires=e.wires, par=[])
+        for i, operation in enumerate(operations):
+            if operation.name == "BasisState":
+                if i > 0:
+                    raise qml.DeviceError(
+                        "The operation BasisState is only supported at the beginning of a circuit."
+                    )
 
-            elif e.name == "PauliY":
-                # Y = (HS^)^.Z.(HS^) and S^=SZ
-                self.apply("PauliZ", wires=e.wires, par=[])
-                self.apply("S", wires=e.wires, par=[])
-                self.apply("Hadamard", wires=e.wires, par=[])
+                self.apply_basis_state(operation)
+            elif operation.name == "QubitStateVector":
+                if i > 0:
+                    raise qml.DeviceError(
+                        "The operation QubitStateVector is only supported at the beginning of a circuit."
+                    )
 
-            elif e.name == "Hadamard":
-                # H = Ry(-pi/4)^.Z.Ry(-pi/4)
-                self.apply("RY", e.wires, [-np.pi / 4])
+                self.apply_qubit_state_vector(operation)
+            else:
+                cirq_operation = self._complete_operation_map[operation.name]
 
-            elif e.name == "Hermitian":
-                # For arbitrary Hermitian matrix H, let U be the unitary matrix
-                # that diagonalises it, and w_i be the eigenvalues.
-                Hmat = e.parameters[0]
-                Hkey = tuple(Hmat.flatten().tolist())
+                # If command is None do nothing
+                if cirq_operation:
+                    cirq_operation.parametrize(*operation.parameters)
 
-                if Hmat.shape not in [(2, 2), (4, 4)]:
-                    raise qml.DeviceError(
-                        "Cirq only supports single-qubit and two-qubit unitary gates and thus only single-qubit and two-qubit Hermitian observables."
+                    self.circuit.append(
+                        cirq_operation.apply(
+                            *[self.qubits[wire] for wire in operation.wires]
+                        )
                     )
 
-                if Hkey in self._eigs:
-                    # retrieve eigenvectors
-                    U = self._eigs[Hkey]["eigvec"]
-                else:
-                    # store the eigenvalues corresponding to H
-                    # in a dictionary, so that they do not need to
-                    # be calculated later
-                    w, U = np.linalg.eigh(Hmat)
-                    self._eigs[Hkey] = {"eigval": w, "eigvec": U}
+        # TODO: get pre rotated state here
+
+        # TODO: Remove duplicate code
```

Comment:
I guess this is a comment for creating a new function something like
```python
def apply_operation(self, operation):
    cirq_operation = self._complete_operation_map[operation.name]

    # If command is None do nothing
    if cirq_operation:
        cirq_operation.parametrize(*operation.parameters)

        self.circuit.append(
            cirq_operation.apply(*[self.qubits[wire] for wire in operation.wires])
        )
```
That could be used starting from line 187 and 204

Reply:
Exactly! I implemented it